spur/core/base.py

```diff
@@ -154,6 +156,26 @@ def as_dict(self):
                     clean["args"][clean_k] = d[k]
         return clean
 
+    def check_usage_eligibility(self, agent: 'Agent'):

```

Comment:
Couple of thoughts:

- Since this is in the Abstract class, it should probably raise `NotImplementedError` (see the `do` method for an example) instead of returning `True`
- Can you add output type hinting? (we should probably do this as best practice everywhere we develop). Something like

   ```python
    def check_usage_elegibility(self, agent: 'Agent') -> bool:
    ```

- This one is mostly style, but I like to name boolean check methods something grammatically matching the question you might ask it, e.g. `available` or `accepting_agents` or something along those lines. Makes it easier to work with when calling in an `if` statement such as `if accepting_agents(): ....`

Reply:
Agreed for the last two points. For the first one, I had it returning `True` mainly because that would be the default behaviour if there isn't any other condition to check aside from the built-in check on the resource capacity. So for any classes inheriting this they don't need to explicitly implement it if they only need the default behaviour. An additional consideration is that for the Block Exclusive Zone implementation, there is going to be some standard shared logic for this method that would be easier to put in the base class to be inherited / extended by children component classes.

Reply:
Also according to the docstrings it seems that the only the `BaseItem` class is explicitly designated as an abstract class ("and should not be used directly"). The `BaseComponent` class inheriting it where the function belongs isn't currently designated as one, though it does have the abstract method `do()`.

Reply:
Ah okay this makes sense then, I must have been thinking we were in the `BaseItem` class. I think it's fine as a base component to have a default method in this case (as it's less likely to be overridden)